fragments/worker.sh

```diff
@@ -109,70 +125,28 @@ DOCKER_CONF=""
 # Start k8s components in containers
 start_k8s() {
     # Start flannel
-    flannelCID=$(docker -H unix:///var/run/docker-bootstrap.sock run \
+    docker run \
         -d \
-        --restart=on-failure \
+        --restart=${RESTART_POLICY} \
         --net=host \
         --privileged \
         -v /dev/net:/dev/net \
+        -v ${FLANNEL_SUBNET_DIR}:/run/flannel \
         ${FLANNEL_IMAGE} \
         /opt/bin/flanneld \
             --ip-masq="${FLANNEL_IPMASQ}" \
             --etcd-endpoints=http://${MASTER_IP}:4001 \
-            --iface="${FLANNEL_IFACE}")
-
-    sleep 10
-
-    echo "flannelCID  ${flannelCID}"
-
-    # Copy flannel env out and source it on the host
-    docker -H unix:///var/run/docker-bootstrap.sock \
-        cp ${flannelCID}:/run/flannel/subnet.env .
-    source subnet.env
-
-    # Configure docker net settings, then restart it
-    case "${lsb_dist}" in
-        centos)
-            DOCKER_CONF="/usr/lib/systemd/system/docker.service"
-            sed -i "/^ExecStart=/ s~$~ --mtu=${FLANNEL_MTU} --bip=${FLANNEL_SUBNET}~" ${DOCKER_CONF}
-            sed -i.bak 's/^\(MountFlags=\).*/\1shared/' ${DOCKER_CONF}
-            systemctl daemon-reload
-            if ! command_exists ifconfig; then
-                yum -y -q install net-tools
-            fi
-            ifconfig docker0 down
-            yum -y -q install bridge-utils && brctl delbr docker0 && systemctl restart docker
-            ;;
-        amzn)
-            DOCKER_CONF="/etc/sysconfig/docker"
-            echo "OPTIONS=\"\$OPTIONS --mtu=${FLANNEL_MTU} --bip=${FLANNEL_SUBNET}\"" | tee -a ${DOCKER_CONF}
-            ifconfig docker0 down
-            yum -y -q install bridge-utils && brctl delbr docker0 && service docker restart
-            ;;
-        ubuntu|debian) # TODO: today ubuntu uses systemd. Handle that too
-            DOCKER_CONF="/etc/default/docker"
-            echo "DOCKER_OPTS=\"\$DOCKER_OPTS --mtu=${FLANNEL_MTU} --bip=${FLANNEL_SUBNET}\"" | tee -a ${DOCKER_CONF}
-            ifconfig docker0 down
-            apt-get install bridge-utils
-            brctl delbr docker0
-            service docker stop
-            while [ `ps aux | grep /usr/bin/docker | grep -v grep | wc -l` -gt 0 ]; do
-                echo "Waiting for docker to terminate"
-                sleep 1
-            done
-            service docker start
-            ;;
-        *)
-            echo "Unsupported operations system ${lsb_dist}"
-            exit 1
-            ;;
-    esac
-
-    # sleep a little bit
-    sleep 5
-
-    # Start kubelet & proxy in container
-    # TODO: Use secure port for communication
+            --iface="${FLANNEL_IFACE}"
+
+	# Wait for the flannel subnet.env file to be created instead of a timeout. This is faster and more reliable
+	local SECONDS=0
+	while [[ ! -f ${FLANNEL_SUBNET_DIR}/subnet.env ]]; do
```

Comment:
It might exist from previous run. We should remove the file before starting flannel or check if it already exists or something else...